prow/pod-utils/clone/clone.go

```diff
@@ -195,13 +195,23 @@ func (g *gitCtx) commandsForBaseRef(refs prowapi.Refs, gitUserName, gitUserEmail
 		commands = append(commands, g.gitCommand("config", "http.cookiefile", cookiePath))
 	}
 
-	if refs.CloneDepth > 0 {
-		commands = append(commands, g.gitFetch(g.repositoryURI, "--tags", "--prune", "--depth", strconv.Itoa(refs.CloneDepth)))
-		commands = append(commands, g.gitFetch("--depth", strconv.Itoa(refs.CloneDepth), g.repositoryURI, refs.BaseRef))
-	} else {
-		commands = append(commands, g.gitFetch(g.repositoryURI, "--tags", "--prune"))
-		commands = append(commands, g.gitFetch(g.repositoryURI, refs.BaseRef))
+	var depthArgs []string
+	if d := refs.CloneDepth; d > 0 {
+		depthArgs = append(depthArgs, "--depth", strconv.Itoa(d))
+	}
+
+	if !refs.SkipFetchHead {
```

Comment:
Since we explicitly fetch the base and HEAD we care about anyways, can't we simply omit fetching the repos head altogether?

Also, terminology is confusing here, becaue the head this refers to is not e.G. the pull head but the repo head

Reply:
I am anticipating this causing problems for repos like k/k which need refs to previous branch tags to exist during testing. Thoughts?